apps/sms/test/unit/compose_test.js

```diff
@@ -1041,48 +1063,64 @@ suite('compose_test.js', function() {
     });
 
     suite('attach button', function() {
-      var request;
       setup(function() {
-        request = {};
-        this.sinon.stub(Compose, 'requestAttachment').returns(request);
+        this.sinon.stub(Compose, 'requestAttachment');
         this.sinon.stub(Utils, 'alert').returns(Promise.resolve());
         this.sinon.stub(Compose, 'append');
         this.sinon.stub(console, 'warn');
-
-        attachButton.click();
       });
 
       test('request an attachment', function() {
+        Compose.requestAttachment.returns(Promise.resolve());
+        attachButton.click();
+
         sinon.assert.called(Compose.requestAttachment);
       });
 
-      test('onsuccess', function() {
+      test('onsuccess', function(done) {
         var attachment = mockAttachment();
-        request.onsuccess(attachment);
-        sinon.assert.calledWith(Compose.append, attachment);
+        var requestPromise = Promise.resolve(attachment);
+        Compose.requestAttachment.returns(requestPromise);
+
+        attachButton.click();
+        requestPromise.then(function() {
+          sinon.assert.calledWith(Compose.append, attachment);
+        }).then(done, done);
       });
 
       suite('onerror,', function() {
-        test('file too large', function() {
-          request.onerror(new Error('file too large'));
-
-          sinon.assert.calledWith(
-            Utils.alert, {
-              id: 'attached-files-too-large',
-              args: { n: 1, mmsSize: '295' }
-            }
+        test('file too large', function(done) {
+          Compose.requestAttachment.returns(
+            Promise.reject(new Error('file too large'))
           );
+          attachButton.click();
+
+          Promise.reject().then(() => {
```

Comment:
Here is one of the variations of hard-to-notice problem with "catch" test I mentioned above, I believe success callback will never be called here and test will always succeed, we need ```catch``` here :) Also you can probably use remembered ```Promise.reject(new Error('file too large'))``` here